src/main/java/fi/metatavu/metamind/bot/BotController.java

```diff
@@ -110,35 +111,50 @@ public MetamindBot getBotInstance(Story story) {
     String storyJson = story.getStoryJson();
     
     Map<String, SlotModel> slotModels = new HashMap<>();
-    IntentModel intentModel = null;
+    List<IntentModel> intentModels = new ArrayList<>();
+    List<MachineLearningConfig> machineLearningConfigs = storyConfig.getMachineLearning();
     
     if (storyConfig.getMachineLearning() != null) {
       loadSlotModels(storyConfig, slotModels);
       
-      String intentModelName = storyConfig.getMachineLearning().getIntentModel();
-      if (intentModelName != null) {
-        intentModel = modelsContoller.findIntentModelByName(String.format("%s.bin", intentModelName));
-        if (intentModel == null && logger.isWarnEnabled()) {
-          logger.warn(String.format("Failed to load intent model %s", intentModelName));
+      List<String> intentModelNames = new ArrayList<>();
+      
+      for (MachineLearningConfig machineLearningConfig : machineLearningConfigs) {
+        intentModelNames.add(machineLearningConfig.getIntentModel());
+      }
+      
+      for (String intentModelName : intentModelNames) {
+        if (intentModelName != null) {
+          IntentModel intentModel = modelsContoller.findIntentModelByName(String.format("%s.bin", intentModelName));
+          
+          if (intentModel == null && logger.isWarnEnabled()) {
+            logger.warn(String.format("Failed to load intent model %s", intentModelName));
+          } else {
+            intentModels.add(intentModel);
+          }
         }
       }
     }
     
-    MetamindBotConfiguration botConfiguration = new MetamindBotConfiguration(storyConfig, functions, storyJson, intentModel, slotModels);
+    MetamindBotConfiguration botConfiguration = new MetamindBotConfiguration(storyConfig, functions, storyJson, intentModels, slotModels);
     
     return new MetamindBot(botConfiguration);
   }
 
   protected void loadSlotModels(StoryConfig storyConfig, Map<String, SlotModel> slotModels) {
```

Comment:
Loading slot models by the collection of slot model names could be moved to another method to make sonar happy but this is fine too in my opinion.